### PR TITLE
Polyhedron_demo: Remove the Remeshing_plugin 

### LINKED ISSUE #2271: Polyhedron_demo: Remaining plugins that needs to work with Surface_mesh
## Issue Details

There are still plugins that need work for use with a Surface_mesh:

- [x] the Remeshing_plugin should be integrated into the Mesh_3_plugin.
- [x] Surf_io_plugin and Surf_to_sm_io_plugin should be merged and depend on the Polyhedron_mode.
- [x] the Fairing_plugin is not touched

### PR DESCRIPTION
**Pull request by maxGimeno**
## Summary of Changes
This PR removes the Remesher_plugin form the Polyhedron_demo and add a feature to the Mesh_3_plugin that allows to output a FaceGraph_item instead of a c3t3_item when performing a surface meshing.

## Release Management
It uses files introduced in #2254.

* Affected package(s): Polyhedron_demo
* Issue(s) solved (if any): partially fixes #2271

### CODE REVIEW

File: Mesh_3/include/CGAL/IO/Complex_3_in_triangulation_3_to_facegraph.h [hunk 1]
```diff
@@ -0,0 +1,133 @@
+#ifndef CGAL_COMPLEX_3_IN_TRIANGULATION_3_TO_FACEGRAPH_H
```
Comment (lrineau) [inline Mesh_3/include/CGAL/IO/Complex_3_in_triangulation_3_to_facegraph.h]: This file must have license and copyright notices. That is the reason of the [Travis job failure][1].

  [1]: https://travis-ci.org/CGAL/cgal/jobs/255958161

File: Surface_mesher/include/CGAL/IO/output_surface_facets_to_facegraph.h [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+// General Public License as published by the Free Software Foundation,
+// either version 3 of the License, or (at your option) any later version.
+//
+// Licensees holding a valid commercial license may use this file in
+// accordance with the commercial license agreement provided with the software.
+//
+// This file is provided AS IS with NO WARRANTY OF ANY KIND, INCLUDING THE
+// WARRANTY OF DESIGN, MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.
+//
+// $URL$
```
Comment (lrineau) [inline Surface_mesher/include/CGAL/IO/output_surface_facets_to_facegraph.h]: You can copy the copyright and license notices from here.

**Comment by sloriot** [comment]
I guess `facegraph` should be replace by `triangle_mesh` to be consistent.

**Comment by maxGimeno** [comment]
Everythong done in #2254 will be done here too. I am just waiting for the small feature to be definitely finished.

**Comment by lrineau** [comment]
TODO: wait for #2254, and then rebase.

**Comment by lrineau** [comment]
#2254 has been merged. You can bow rebase.

**Comment by maxGimeno** [comment]
rebased

**Comment by lrineau** [comment]
Conflicts with `master`.

**Comment by maxGimeno** [comment]
rebased.

**Comment by lrineau** [comment]
```
Compiling remeshing_plugin ... 
make[1]: Entering directory `/home/cgal_tester/build/src/cmake/platforms/CentOS7-Release/test/Polyhedron_Demo'
make[1]: *** No rule to make target `remeshing_plugin'.  Stop.
```
https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-4.12-Ic-151/Polyhedron_Demo/TestReport_lrineau_CentOS7-Release.gz

Probably you forgot to remove the line from the `cgal_test` shell script.

**Comment by lrineau** [comment]
I have tested the modification on my machine. It works well, but I have the impression that is created the face graph with the wrong orientation. Or maybe a random orientation. I have only tested it once.

**Comment by maxGimeno** [comment]
You are right, I'll use #2545 to fix it

**Comment by lrineau** [comment]
Tested in [CGAL-4.12-Ic-152], successfully, and then merged.

  [CGAL-4.12-Ic-152]: https://cgal.geometryfactory.com/CGAL/testsuite/results-4.12-Ic-152.shtml

**Pull request merged**